### PR TITLE
Bugfix/all variables injected when function has no variables

### LINKED ISSUE #337: All variables from process are injected into a task function that takes in no parameters
**Describe the bug**
When you create a function that doesn't take any parameters in its signature, all the variables from the process scope are passed into it.
The workaround I used to solve this is to take in `*args, **kwargs` inside the task function signature.

**To Reproduce**
1. Create a task function that takes in no parameters.
2. Run a process which uses that task in it.
3. Code crashes from an unexpected parameters error.

**Expected behavior**
No variables should be injected and the function should complete normally.

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Version 3.0.4

### PR DESCRIPTION
**Pull request by ohasi**
Fix bug where function decorated as task receives all variables that are available to the task causing an `unexpected keyword argument` error. Referenced in #337.

The bug originates from the zeebe grpc API for `ActivateJobs`. If no variables are passed to `fetchVariable `, it fetches all available variables.

## Changes

- Add a `include_variables` flag to private function `_create_job_from_raw_job`. If flag is set to False, retrieved variables will be ignored. 
The function is called in one place, which now calls it with `bool(variables_to_fetch)` as the flag's value. `None` or an empty list will cause no variables to be injected to the `Job` object, so they will never end up getting passed to the function.

### New Features *(required)*

- Support for functions with no arguments as task functions

## Checklist

- [x] Unit tests
- [x] Documentation

## References

[ActivateJobs API](https://docs.camunda.io/docs/apis-tools/grpc/#activatejobs-rpc)

Fixes #337

### CODE REVIEW

**Comment by JonatanMartens** [comment]
Looks good, but please add a simple unit test so that this bug doesn't reproduce.

**Comment by ohasi** [comment]
> Looks good, but please add a simple unit test so that this bug doesn't reproduce.

Done.
Something to have in mind that someone mentioned to me - this pr will remove the ability to intentionally get all parameters by setting the `variables_to_fetch` to `[]` or `None`. 
I don;t think that's a problem or intended functionality, but just making sure

File: pyzeebe/worker/task_router.py [hunk 1]
```diff
@@ -71,7 +72,7 @@ def task(
         def task_wrapper(task_function: Callable):
             config = TaskConfig(
                 task_type,
-                exception_handler,
+                exception_handler or self._default_exception_handler,
```
Comment (JonatanMartens) [inline pyzeebe/worker/task_router.py:75]: Would be nice to add  a small unit test for this

File: tests/unit/utils/random_utils.py [hunk 2]
```diff
@@ -27,8 +28,15 @@ def random_job(
...
         custom_headers={},
         deadline=randint(0, RANDOM_RANGE),
         zeebe_adapter=zeebe_adapter,
     )
+
+
+def _build_dict_from_list(keys: List[str], *values):
+    if not values:
+        return dict(zip(keys, keys))
+    assert len(keys) == len(values)
```
Comment (JonatanMartens) [inline tests/unit/utils/random_utils.py:41]: No need to add this imo

File: tests/unit/utils/random_utils.py [hunk 3]
```diff
@@ -27,8 +28,15 @@ def random_job(
...
         element_instance_key=randint(0, RANDOM_RANGE),
-        variables={},
+        variables=task.config.variables_to_fetch or _build_dict_from_list(list(str(uuid4()))),
         custom_headers={},
         deadline=randint(0, RANDOM_RANGE),
         zeebe_adapter=zeebe_adapter,
     )
+
+
+def _build_dict_from_list(keys: List[str], *values):
```
Comment (JonatanMartens) [inline tests/unit/utils/random_utils.py:38]:
```suggestion
def _build_dict_from_list(keys: List[str], *values) -> dict:
```

**Review by JonatanMartens — CHANGES_REQUESTED** [review]

File: tests/unit/utils/random_utils.py (hunk 2 above)
Comment (ohasi) [reply tests/unit/utils/random_utils.py:41]: Removed :)

File: pyzeebe/worker/task_router.py (hunk 1 above)
Comment (ohasi) [reply pyzeebe/worker/task_router.py]: I'll add it in #349

**Comment by ohasi** [comment]
Waiting for re-review :)

**Review by ohasi — COMMENTED** [review]
Waiting for re-review :)

**Comment by dimastbk** [comment]
fixed in #404

**Pull request closed without merging**